packages/stargate/src/signingstargateclient.ts

```diff
@@ -323,18 +342,28 @@ export class SigningStargateClient extends StargateClient {
       };
     }
 
-    return isOfflineDirectSigner(this.signer)
+    return this.getSigningMode() == SigningMode.Direct
       ? this.signDirect(signerAddress, messages, fee, memo, signerData)
       : this.signAmino(signerAddress, messages, fee, memo, signerData);
   }
 
-  private async signAmino(
+  /**
+   * Returns the signing mode to be used.
+   * If you need another way to get the signing mode to be used (maybe calling a custom method
+   * of your signer), then you can simply override this.
+   * @protected
+   */
+  protected getSigningMode(): SigningMode {
+    return isOfflineDirectSigner(this.signer) ? SigningMode.Direct : SigningMode.Amino;
+  }
```

Comment:
This problem came up with Keplr before. The type itself is insufficient to determine the sign mode side it could be either a software wallet or a Ledger behind Keplr.

Woudn't it make sense to extend the `OfflineSigner` in a way that the sign mode can be determined at runtime instead of compine time?

Would this avoid that clients have to change the logic of this function?